libp2p/Host.cpp

```diff
@@ -149,12 +149,22 @@ void Host::stop()
 		return;
 
 	// wait for m_timer to reset (indicating network scheduler has stopped)
-	while (!!m_timer)
-		this_thread::sleep_for(chrono::milliseconds(50));
+	{
+		std::unique_lock<std::mutex> l{x_runTimer};
+		while (!!m_timer)
+		{
+			l.unlock();
+			this_thread::sleep_for(chrono::milliseconds(50));
+			l.lock();
+		}
+	}
 
 	// stop worker thread
 	if (isWorking())
 		stopWorking();
+
+	// This destroyes EthereumHost objects.  EthereumHost objects contain a references to Clients' members, so EthereumHost objects need to disappear first.
+	m_capabilities.clear();
```

Comment:
I think we can't put this into `Host::stop()`. Network can be stopped and then restarted with admin RPC. (`Host::stop()` & `Host::start()` are called in this case and restart won't work)

Reply:
This change is not necessary anymore.  I don't see TSAN warnings without it.  I'll remove it.

Reply:
So it's still here, are we removing it?

Reply:
Yes.  Done.